url-shortener/common/src/Common/Route.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE EmptyCase #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Common.Route where
+
+import Data.Text (Text)
+import qualified Data.Text as T
+import Data.Function
+import Data.Functor.Identity
+import Database.Id.Class (Id)
+import Database.Id.Obelisk.Route (idPathSegmentEncoder)
+
+import Obelisk.Route
+import Obelisk.Route.TH
+
+import Common.Schema
+
+data BackendRoute :: * -> * where
+  BackendRoute_Missing :: BackendRoute ()
+  BackendRoute_Get_Url :: BackendRoute (Id Url)
```

Comment:
Let's call this `BackendRoute_GetUrl` just to be more idiomatic with the rest of our apps.